conda/deprecations.py

```diff
@@ -124,6 +125,50 @@ def inner(*args, **kwargs):
 
         return deprecated_decorator
 
+    def action(
+        self,
+        deprecate_in: str,
+        remove_in: str,
+        action: Action,
+        *,
+        addendum: str | None = None,
+        stack: int = 0,
+    ):
+        class DeprecationMixin:
+            def __init__(inner_self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+
+                category, message = self._generate_message(
+                    deprecate_in,
+                    remove_in,
+                    (
+                        # option_string are ordered shortest to longest,
+                        # use the longest as it's the most descriptive
+                        f"`{inner_self.option_strings[-1]}`"
+                        if inner_self.option_strings
+                        # if not a flag/switch, use the destination itself
+                        else f"`{inner_self.dest}`"
+                    ),
+                    addendum=addendum,
+                )
+
+                # alert developer that it's time to remove something
+                if not category:
+                    raise DeprecatedError(message)
```

Comment:
I'm a little curious why this isn't being covered by tests. Isn't the point of this function to eventually trigger such a state? Seems like it would be  pretty import to make sure this line is covered.

Reply:
You're right, added applicable tests